File: packages-gen/js/ui/state/crud.js
const form = require("@xtalk/ui/state/form.js")

const collection = require("@xtalk/ui/state/collection.js")

function spec(id,models,fields,columns,actions,opts){
  return {
    "id":id,
    "strategy":"page_controller",
    "models":models || {},
    "fields":fields || [],
    "columns":columns || [],
    "actions":actions || [],
    "opts":opts || {}
  };
}

function create_state(crud_spec,values){
  return {
    "status":"idle",
    "spec":crud_spec,
    "collection":collection.create({}),
    "form":form.create(values || {},{}),
    "record":null,
    "mode":"list",
    "errors":{}
  };
}

function set_modef(state,mode,record){
  state["mode"] = mode;
  state["record"] = record;
  return state;
}

module.exports = {
  ["spec"]:spec,
  ["create_state"]:create_state,
  ["set_modef"]:set_modef
}

File: packages-gen/dart/libs/xt-ui/lib/state/crud.dart
import 'package:xtalk_ui/state/form.dart' as form;
import 'package:xtalk_ui/state/collection.dart' as collection;



spec(id, models, fields, columns, actions, opts) {
  return <dynamic, dynamic>{
    "id":id,
    "strategy":"page_controller",
    "models":models ?? <dynamic, dynamic>{},
    "fields":fields ?? <dynamic>[],
    "columns":columns ?? <dynamic>[],
    "actions":actions ?? <dynamic>[],
    "opts":opts ?? <dynamic, dynamic>{}
  };
}

create_state(crud_spec, values) {
  return <dynamic, dynamic>{
    "status":"idle",
    "spec":crud_spec,
    "collection":collection.create(<dynamic, dynamic>{}),
    "form":form.create(values ?? <dynamic, dynamic>{},<dynamic, dynamic>{}),
    "record":null,
    "mode":"list",
    "errors":<dynamic, dynamic>{}
  };
}

set_modef(state, mode, record) {
  state["mode"] = mode;
  state["record"] = record;
  return state;
}